proposed/http-middleware/middleware-meta.md

```diff
@@ -171,6 +170,33 @@ that are based around StackPHP.
 The double pass approach is much newer but has already been widely adopted by
 early adopters of HTTP Messages.
 
+
+### 4.4 Chosen Approach
+
+The double-pass approach has some significant issues regarding implementation.
+
+The most severe is that passing an empty response has no guarantees that the
+response is in a usable state. This is further exacerbated by the fact that a
+middleware may modify the response before passing it for further dispatching.
+
+Further compounding the problem is that there is no way to ensure that the
+response body has not been written to, which can lead to incomplete output or
+error responses being sent with cache headers attached.
+
+Some have argued that passing the response helps ensure dependency inversion.
+While this it is true that it helps avoid depending on a specific implementation
+of HTTP messages, the problem can also be resolved by injecting factories into
+the middleware to create HTTP message objects. With the creation of HTTP Factories
+in PSR-17, a standard approach to handling dependency inversion is possible.
+
+Using lambda approach ensures a more consistent and predictable developer
+experience, which helps prevent incorrect or less optimal implementation.
+The lambda approach is also somewhat easier to explain and has wider overall
+adoption within the ecosystem due to existing implementations.
+
+Using interface declarations instead of `callable` eliminates some flexibility
+in favor of better runtime safety and debugging.
```

Comment:
I'd add some narrative here about the fact that it also allows existing implementations to adapt in a way that keeps their own backwards compatibility, while providing forwards compatibility with this proposal.